fmriprep/interfaces/cifti.py

```diff
@@ -0,0 +1,287 @@
+        """Assign arbitrary label to combination of CIFTI spaces"""
+        space = None
+        variants = {
+            # to be expanded once addtional spaces are supported
+            'space1': ['fsaverage5', 'MNI152NLin2009cAsym'],
+            'space2': ['fsaverage6', 'MNI152NLin2009cAsym'],
+        }
+        for sp, targets in variants.items():
+            if all(target in targets for target in
+                    [self.inputs.surface_target, self.inputs.volume_target]):
+                space = sp
+        if space is None:
+            raise NotImplementedError
+
+        variant_key = os.path.abspath('dtseries_variant.json')
+        with open(variant_key, 'w') as fp:
+            json.dump({space: variants[space]}, fp)
+        return variant_key, space
+
+    def _fetch_data(self):
+        """Converts inputspec to files"""
+        if (self.inputs.surface_target == "fsnative" or
+                self.inputs.volume_target != "MNI152NLin2009cAsym"):
+            # subject space is not support yet
+            raise NotImplementedError
+
+        annotation_files = sorted(glob(os.path.join(self.inputs.subjects_dir,
+                                                    self.inputs.surface_target,
+                                                    'label',
+                                                    '*h.aparc.annot')))
+        if not annotation_files:
+            raise IOError("Freesurfer annotations for %s not found in %s" % (
+                          self.inputs.surface_target, self.inputs.subjects_dir))
+
+        label_space = 'oasis_dkt31_mni152'
+        label_template = os.path.dirname(getters.get_dataset(label_space))
+        label_file = os.path.join(label_template,
+                                  'oasis_trt20_DKT31-MNI152NLin',
+                                  ('OASIS-TRT-20_jointfusion_DKT31_CMA_labels'
+                                   '_in_MNI152NLin2009cAsym_2mm_v2.nii.gz'))
+
+        download_link = getters.OSF_PROJECT_URL + getters.OSF_RESOURCES[label_space][0]
+        return annotation_files, label_file, download_link
+
+    @staticmethod
+    def _create_cifti_image(bold_file, label_file, annotation_files, gii_files,
+                            volume_target, surface_target, tr, download_link=None):
+        """
+        Generate CIFTI image in target space
+
+        Parameters
+            bold_file : 4D BOLD timeseries
+            label_file : label atlas
+            annotation_files : FreeSurfer annotations
+            gii_files : 4D BOLD surface timeseries in GIFTI format
+            volume_target : label atlas space
+            surface_target : gii_files space
+            tr : repetition timeseries
+            download_link : URL to download label_file
+
+        Returns
+            out_file : BOLD data as CIFTI dtseries
+        """
+
+        bold_img = nb.load(bold_file)
+        label_img = nb.load(label_file)
+        if not bold_img.shape[:3] == label_img.shape:
```

Comment:
You can drop this check. `nilearn.image.resample_to_img` will just return a copy if the shape and affine match.

So I'd actually write the whole loading section in two lines:

```Python
label_img = nb.load(label_file)
bold_img = resample_to_img(bold_file, label_img)
```